frameit/lib/frameit/device_types.rb

```diff
@@ -134,9 +134,9 @@ module Devices
     IPHONE_13_PRO_MAX ||= Frameit::Device.new("iphone13-pro-max", "Apple iPhone 13 Pro Max", 11, [[1284, 2778], [2778, 1284]], 458, Color::GRAPHITE, Platform::IOS)
     IPHONE_13_MINI ||= Frameit::Device.new("iphone-13-mini", "Apple iPhone 13 Mini", 11, [[1080, 2340], [2340, 1080]], 476, Color::MIDNIGHT, Platform::IOS)
     IPHONE_14 ||= Frameit::Device.new("iphone-14", "Apple iPhone 14", 12, [[1170, 2532], [2532, 1170]], 460, Color::MIDNIGHT, Platform::IOS)
-    IPHONE_14_PLUS ||= Frameit::Device.new("iphone-14-plus", "Apple iPhone 14 Plus", 12, [[1284, 2778], [2778, 1284]], 458, Color::MIDNIGHT, Platform::IOS)
-    IPHONE_14_PRO ||= Frameit::Device.new("iphone-14-pro", "Apple iPhone 14 Pro", 12, [[1178, 2556], [2556, 1178]], 460, Color::PURPLE, Platform::IOS)
-    IPHONE_14_PRO_MAX ||= Frameit::Device.new("iphone14-pro-max", "Apple iPhone 14 Pro Max", 12, [[1290, 2796], [2796, 1290]], 458, Color::PURPLE, Platform::IOS)
+    IPHONE_14_PLUS ||= Frameit::Device.new("iphone14-plus", "Apple iPhone 14 Plus", 12, [[1284, 2778], [2778, 1284]], 458, Color::MIDNIGHT, Platform::IOS)
+    IPHONE_14_PRO ||= Frameit::Device.new("iphone-14-pro", "Apple iPhone 14 Pro", 12, [[1179, 2556], [2556, 1179]], 460, Color::PURPLE, Platform::IOS)
```

Comment:
The resolution in `master` is incorrect. This is the correct resolution [1179, 2556].

Reply:
@mrvincenzo the resolution on master is correct! I‘ve tested it and generated all production screens with it.

Reply:
@sathoeni what your are saying is inaccurate - iPhone 14 Pro resolution in `master` is incorrect. The fact that it works, doesn't make it correct.

[iPhone 14 Pro and Pro Max spec](https://www.apple.com/ge/iphone-14-pro/specs/)

## iPhone 14 Pro
<img width="300" alt="image" src="https://github.com/fastlane/fastlane/assets/5012608/584cdb58-fe0e-4251-8743-43a54616956a"> 
 
The resolution should be `2556x1179` and not `2556x1178` like in `master`. 
You can also take a screenshot of the iPhone 14 Simulator and check its size. You will discover it is `2556x1179`.

## iPhone 14 Pro Max
<img width="300" alt="image" src="https://github.com/fastlane/fastlane/assets/5012608/3b5b9cd9-291a-4879-b9c0-c6fb428f1922">

The DPI should be 460 and not 458 like in `master`.

Reply:
Thanks @mrvincenzo for the screenshots. Looks like our CI and my local machine produces somehow screenshots with another resolution.
Can you also adapt the size of the screen in the following repo?
https://github.com/fastlane/frameit-frames

Reply:
PR to frameit-frames: https://github.com/fastlane/frameit-frames/pull/34

Reply:
is there a way to force `fastlane frameit download_frames` download the frames from my forked repo instead of the default one?

Reply:
@mrvincenzo 
I found the issue regarding the screensize
When you take a manual screenshot from the simulator the width of the iPhone 14 Pro is 1179. When you take a screenshot with an UITest (XCUIScreen.main.screenshot()) then it's width is 1178.
I assume it's an XCode bug, I will open a ticket for apple.

We should reopen your issue, I'm sure @getaaron  can merge it later, after the issue is fixed by xcode. In the meantime we should keep 1178, so it's works when using fastlane snapshot.

Reply:
I've opened FB13569779.

with this text:
When using XCUIScreen.main.screenshot() or XCUIApplication().screenshot for generating screenshots for the iPhone 14 Pro the screensize is different (and according to documentation) not correct, compared the taking screenshots manually from the simulator CTRL+S When I take a screenshot manually from the simulator the screensize is 1179 × 2556 pixels When I take a screenshot with XCUIScreen.main.screenshot() the screensize is 1178 × 2556 pixels According to the documentation the screenshot size should be 1179 × 2556! https://developer.apple.com/help/app-store-connect/reference/screenshot-specifications

and following evidence

![XCUIScreenshot](https://github.com/fastlane/fastlane/assets/5508646/372c04f4-75c9-4116-9bb6-a2d20d4a4fbb)
![manual screenshot](https://github.com/fastlane/fastlane/assets/5508646/d91c5be5-8b58-4bb0-8c84-0cf758e8851f)

Reply:
I agree — I believe we should separate these 2 issues:

- Adding iPhone 14 device lineup + new device colors, which is mostly what this PR is doing
- Fixing the size of the image of the iPhone 14 Pro, which apparently requires other changes (e.g. https://github.com/sathoeni/fastlane/commit/c5b57b2a54606060c5b25ff86021cf93a854453d) plus perhaps changes in the frames in https://github.com/fastlane/frameit-frames (uncertain, to me)

If we proceed with this PR as is, it looks like it will cause regressions if we don't address the other co-related changes simultaneously? Looks like we'd be falling into scope creep, no?

What do you guys think? @getaaron @mrvincenzo @sathoeni

Reply:
@rogerluan What do you think? Can we proceed on this one?